apps/widget-configurator/src/app/configurator/controls/TokenListControl.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState, useCallback, useMemo } from 'react'
+
+import {
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Button,
+  OutlinedInput,
+  InputLabel,
+  ListItemText,
+  MenuItem,
+  FormControl,
+  Select,
+  TextField,
+  SelectChangeEvent,
+  Chip,
+  Box,
+} from '@mui/material'
+
+import { TOKEN_LISTS } from '../consts'
+
+const ITEM_HEIGHT = 48
+const ITEM_PADDING_TOP = 8
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+}
+
+type TokenListControlProps = {
+  onTokenListSelect: (selectedUrls: string[]) => void
+  initialSelectedTokenLists: string[]
+}
+
+type CustomList = {
+  name: string
+  url: string
+}
+
+type AddCustomListDialogProps = {
+  open: boolean
+  onClose: () => void
+  onAdd: (newList: CustomList) => void
+}
+
+const AddCustomListDialog = ({ open, onClose, onAdd }: AddCustomListDialogProps) => {
+  const [customList, setCustomList] = useState<CustomList>({ name: '', url: '' })
+  const [errors, setErrors] = useState({ name: false, url: false })
+
+  const validateName = (name: string) => {
+    return name.trim() !== ''
+  }
+
+  const validateURL = (url: string) => {
+    const pattern = new RegExp(
+      '^(https?:\\/\\/)?' + // protocol
+        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+        '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+        '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+        '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+        '(\\#[-a-z\\d_]*)?$',
+      'i'
+    ) // fragment locator
+    return !!pattern.test(url)
+  }
```

Comment:
Is the logic in this file duplicated from somewhere?
I feel like we already have those validations somewhere

Reply:
Fixed